api/v1alpha1/hardwareclassification_types.go

```diff
@@ -48,6 +48,31 @@ type HardwareCharacteristics struct {
 	Nic *Nic `json:"nic,omitempty"`
 	// +optional
 	Ram *Ram `json:"ram,omitempty"`
+	// +optional
+	Firmware *Firmware `json:"firmware,omitempty"`
+	// +optional
+	SystemVendor *SystemVendor `json:"systemVendor,omitempty"`
+}
+
+// Firmware contains firmware details extracted from the hardware profile
+type Firmware struct {
```

Comment:
Do we need this to be 2 levels if there's only one member in this struct? What other fields do we anticipate adding here?

Reply:
We tried to keep it same as bmh, but now we have changed it to single level only.

Reply:
Oh, sorry, I didn't realize that's what you were doing!

If we're going to add more fields to the firmware struct in BMH, we may also want those here, so I think it makes sense to keep them the same.